[DEV][FEATURES] Done all workflow of app, all data using Firestore, Chat real-time, Local Notification, Firebase Hosting

dinhthongchau opened the pull request:
✅ Requirements:

- UI accuracy: 95%+ match with Figma
- All features: fully functional and performant
- Responsive design for multiple screen sizes
- Clean Architecture:
    +data layer: data using Firebase
    +domain layer: define interfaces & logic -> work fine
    +presentation layer: manage UI state with BloC (The authentication fles has not changed to BloC yet; it is still using setState.)

✅ Others:
- Deploy app with Firebase Hosting http://biwofinance.web.app/
- Able to edit and remove a transaction
- Add more UI, functions on SearchScreen, CalendarScreen, CategoriesScreen to more logic when using app.
- Customer Service: Chat real-time with helper(real people).
- Notification when received message from helper
- Flows to Implement: done all ( Splash Screen → Onboarding → Authentication → Home → Account Balance → Create Transaction → Notifications → Transaction History → Analysis → Category → My Profile( edit profile,change password, delete account. ) ) .

dinhthongchau committed:
fix: Resolved comments

dinhthongchau committed:
feat: Build categories screens

dinhthongchau committed:
feat: Build common utils, bottom navigation, app bar

dinhthongchau committed:
feat: Build transaction UI and functions

dinhthongchau committed:
feat: Build analysis UI and functions

dinhthongchau committed:
feat: Build notifications UI and functions

dinhthongchau committed:
Refactor: Improve spacing in analysis screen and chart display logic

This commit refactors the analysis screen by:

- Utilizing the `spacing` property in the `Column` widget for better vertical spacing between elements.
- Updating the chart display logic to show "No data available" when all chart data points have zero income and expense, in addition to when the chart data is empty.
- Adjusting the Y-axis interval calculation to default to 1000 if `maxYValue` is 0 to prevent division by zero errors.

dinhthongchau committed:
Refactor: Add notifications for transaction operations

This commit introduces notifications for adding, editing, and deleting transactions.

- Moved notification creation for adding transactions from `AddTransactionScreen` to the `TransactionSuccess` state in `AddTransactionScreen`.
- Added notifications for deleting and editing transactions in `transaction_reused_widgets.dart`.
- Ensured notifications include relevant information like title, subtitle, time, and date.

dinhthongchau committed:
Refactor: Improve transaction loading and UI responsiveness

- Optimized transaction loading by initializing with an empty state and allowing month-specific loading.
- Refined UI updates during transaction loading states for better user experience.
- Adjusted mock data generation delay for faster development feedback.
- Minor code cleanup and formatting in various files.

dinhthongchau committed:
Refactor: Improve transaction loading and UI responsiveness

- Optimized transaction loading by initializing with an empty state and allowing month-specific loading.
- Refined UI updates during transaction loading states for better user experience.
- Adjusted mock data generation delay for faster development feedback.
- Minor code cleanup and formatting in various files.

dinhthongchau committed:
chore: remove comment in routes.dart

dinhthongchau committed:
Revert "chore: remove comment in routes.dart"

This reverts commit 590abe2c5b0bb8f8767c164085b9fb62c237e1ae.

dinhthongchau committed:
Refactor: Introduce BLoC for CalendarScreen

This commit refactors the `CalendarScreen` to use a BLoC (`CalendarBloc`) for managing its state and business logic.

Key changes:

-   **New BLoC:** `CalendarBloc` is created to handle:
    -   Loading transactions.
    -   Managing selected date, current month, chart type, and all-month view.
    -   Generating chart data based on the current state.
-   **New Events:** `CalendarEvent` and its subclasses are defined to trigger state changes in the BLoC (e.g., `SelectDateEvent`, `ChangeMonthEvent`, `LoadCalendarTransactionsEvent`).
-   **New States:** `CalendarState` and its subclasses (`CalendarInitial`, `CalendarLoading`, `CalendarSuccess`, `CalendarError`) represent the different states of the calendar screen.
-   **`CalendarScreen` Refactor:**
    -   The `CalendarScreen` is now a `StatelessWidget`.
    -   It uses `BlocConsumer` to listen to state changes and rebuild the UI accordingly.
    -   Local state management (e.g., `_selectedDate`, `_currentMonth`, `_chartData`) is removed and handled by the BLoC.
    -   Methods for generating chart data and handling UI interactions are now dispatched as events to the BLoC.
-   **Error Handling:** The `CalendarError` state is introduced to display error messages.
-   **Shared Data:** The `ChartSampleData` and `ChartType` enum are now defined in `shared_data.dart` (though this file is not part of the diff, it's implied by the imports).

This refactoring improves the separation of concerns, makes the `CalendarScreen` more testable, and centralizes the state management for better maintainability.

dinhthongchau committed:
feat: implement CategoryBloc for category management

This commit introduces the `CategoryBloc` to manage category states and events (load, add, update, delete).

Key changes:
- Created `CategoryBloc`, `CategoryEvent`, and `CategoryState`.
- Integrated `CategoryBloc` into `CategoryListScreen` for fetching and displaying categories.
- Updated `CategoryRepository` to support asynchronous operations (add, update, delete) and return `Future`.
- Implemented add, edit, and delete functionality for categories in `CategoryListScreen` using `CategoryBloc`.
- Fixed a display issue in `CategoryDetailScreen` for negative total balance.
- Refactored `CategoryListScreen` to use `BlocBuilder` and `BlocConsumer` for state management and UI updates.

dinhthongchau committed:
Feat: Implement custom BLoC observer with color-coded logs and enhanced state formatting

This commit introduces a custom `MyBlocObserver` that enhances the debugging experience for BLoC state management. Key features include:

- **Color-coded logs:** Different log types (create, event, change, transition, error, close, debug) are now printed with distinct colors and icons in the console for easier visual parsing.
- **Improved state formatting:** The `_formatStateForLog` helper function is implemented to provide more concise and readable representations of BLoC states in logs. This includes:
    - Summarizing lists of model instances (e.g., `List<SomeModel> (count: 5)`) instead of printing each instance.
    - Abbreviating verbose map content to `{...}` if it exceeds a certain length.
    - Truncating overly long state strings to a maximum display length.
- **General code enhancements and bug fixes:**
    - Updated `TransactionBloc` to fetch updated transactions from the repository after adding a new transaction.
    - Adjusted routing for `CategoryDetailScreen` and `CategoryDetailSaveScreen` to use placeholder data for testing in Flutter Inspector and allow null `moneyType` in `CategoryListScreen`.
    - Added `CategoryBloc` and `CalendarBloc` to the main application's `MultiBlocProvider`.
    - Corrected total balance display in `TransactionScreen` and `HomeScreen` to show negative sign correctly.
    - Updated icons in `CalendarScreen` and `SearchScreen`.
    - Added `formatCurrency` utility function in `NumberFormatUtils`.
    - Improved loading indicator handling in `BottomNavigationBarScaffold` by ensuring `LoadingUtils.showLoading` is called within `addPostFrameCallback` and hiding the loader on error.
    - Added spacing in `HomeScreen` and `SearchScreen` for better layout.

dinhthongchau committed:
Refactor: Add AddTransactionBloc for managing transaction creation state

This commit introduces a new BLoC, `AddTransactionBloc`, to handle the state management for the "add transaction" screen. This refactoring separates the concerns of transaction creation logic from the UI, leading to a cleaner and more maintainable codebase.

The `AddTransactionBloc` is responsible for:
- Loading available categories based on the selected money type.
- Managing the selected date for the transaction.
- Handling the selection of a category.
- Providing the necessary state (selected date, money type, selected category, available categories) to the UI.

The `AddTransactionScreen` has been updated to use this new BLoC, delegating state management tasks to it. This includes:
- Initializing the BLoC with the initial money type and category.
- Dispatching events to the BLoC for loading categories, selecting a date, and selecting a category.
- Building the UI based on the state emitted by the BLoC.

Additionally, the transaction list screen (`TransactionScreen`) header was simplified by removing the notification icon, as it was deemed redundant with the notification icon present in the `AddTransactionScreen`'s app bar. The `TransactionBloc` and related files have been updated to reflect changes in model imports due to the refactoring.

dinhthongchau committed:
Refactor: To Bloc, move analysis models and screen to dedicated files

This commit refactors the analysis feature by:

- Moving the analysis data models (BaseAnalysis, DailyAnalysis, WeeklyAnalysis, MonthlyAnalysis, YearAnalysis) from `analysis_screen.dart` to a new file `lib/data/model/analysis_model.dart`.
- Moving the `AnalysisScreen` widget from `lib/presentation/screens/analysis/analysis_screen.dart` to a new file `lib/presentation/screens/analysis/analysis/analysis_screen.dart`.
- The original `analysis_screen.dart` has been deleted.

This change improves code organization and separation of concerns.

dinhthongchau committed:
Refactor: Introduce HomeBloc and SearchBloc

This commit introduces two new Blocs: `HomeBloc` and `SearchBloc`.

**HomeBloc:**
- Manages the state of time filters (daily, weekly, monthly) on the home screen.
- Includes `HomeEvent` and `HomeState` for event and state management.
- `SelectTimeFilterEvent` is used to change the selected time filter.

**SearchBloc:**
- Manages the search and filtering functionality on the search screen.
- Includes `SearchEvent` and `SearchState`.
- Handles events for applying filters, selecting categories, dates, and report types.
- Filters transactions based on query, selected category, date, and report type.
- Provides loading, success, and error states for search operations.

**Refactoring in Calendar:**
- Renamed `ChartType` enum to `ChartTypeCalendar` to avoid naming conflicts.
- Renamed `SelectDateEvent` to `SelectDateCalendarEvent`.

**Refactoring in Home Screen:**
- Integrated `HomeBloc` to manage time filter selection.
- Removed local state management for time filters.

**Refactoring in Search Screen:**
- Integrated `SearchBloc` to manage search and filtering logic.
- Removed local state management for filters and filtered transactions.
- Updated UI to reflect changes based on `SearchState`.
- Renamed `ReportType` enum to `ReportTypeSearch`.

dinhthongchau committed:
Refactor: Move enum and enhance export structure

- Introduce `enums.dart` for centralized enum definitions.
- Consolidate common exports into `shared_data.dart`.
- Update import paths across various files to reflect these changes.
- Refactor `MyBlocObserver` for improved logging clarity and formatting.
- Minor adjustments to UI elements and dialog handling.

dinhthongchau committed:
Refactor: Introduce Mixins for screens

dinhthongchau committed:
Refactor: Enhance error handling and data presentation

This commit introduces several improvements related to error handling, data display, and code structure:

- **Error Handling:**
    - `TransactionRepository`: Throws an exception if loading transactions fails, providing more specific error information.
    - `common_functions.dart`: Updated `showErrorDialog` to display a default error message if `errorMessage` is null.
    - `transaction_reused_widgets.dart`: Added a try-catch block in `TransactionDetailDialog` when parsing `editedDate` to show a snackbar for invalid date formats.

- **Data Presentation:**
    - `analysis_model.dart`: Changed `MonthlyAnalysis.xValue` to use `startDate.year` for correct month display.
    - `transaction_screen_mixin.dart`: Modified the month selection `PopupMenuButton` to display "No months available" if `availableMonths` is empty.

- **Code Refactoring:**
    - `search_screen.dart`: Removed redundant `MultiBlocProvider` and `BlocProvider.value` as the necessary Blocs are likely provided higher up in the widget tree.
    - `main.dart`: Commented out an unused `AddTransactionBloc` provider.

ThinhSuy1 committed:
Authentication flow: Sign up, Log in, Edit profile, change password, Delete Account, Help Center with chatbot and Helper, Using Firebase

dinhthongchau committed:
Refactor: Remove duplicate asset path in pubspec.yaml

This commit removes a redundant entry for `assets/IconComponents/` from the `assets` section in `pubspec.yaml`.

dinhthongchau committed:
Merge branch 'features/authentication-profile' into features/authen-trans-2905

dinhthongchau committed:
Feat: Implement NotificationBloc for chat notifications

This commit introduces a `NotificationBloc` to manage the state of new chat messages.

- **NotificationBloc:**
  - Manages a `NotificationState` which holds a `hasNewMessage` boolean.
  - Handles `NotificationReceived` events to set `hasNewMessage` to true.
  - Handles `NotificationRead` events to set `hasNewMessage` to false.

- **NotificationScreen:**
  - Displays a list of chat rooms for the current user.
  - Filters chat rooms to only show those with messages.
  - Tapping a chat room navigates to the respective chat screen.
  - The UI is built using `StreamBuilder` for real-time updates from Firebase and `FutureBuilder` to filter rooms with messages.
  - Includes a styled AppBar and list items for notifications.

dinhthongchau committed:
refactor: format files

dinhthongchau committed:
Refactor: Update navigation for online support screens

This commit refactors the navigation for the online support screens.

- The route for `/profile-online-support-helper-chat` has been commented out in `routes.dart`.
- Navigation to `ProfileOnlineSupportAiScreen` from `ProfileOnlineSupportAiLobbyScreen` now uses `Navigator.push` with `MaterialPageRoute` instead of `context.go`, passing `chatHistory` as an argument to the screen's constructor.
- The route definition for `/profile-online-support-ai` in `routes.dart` has been updated to no longer expect `extra` data.
- Navigation to `ProfileOnlineSupportHelperChatScreen` from `ProfileOnlineSupportHelperScreen` now uses `Navigator.push` with `MaterialPageRoute` instead of `context.go`, passing `chatRoom` and `helperId` as arguments to the screen's constructor.

Additionally, the default email and password in `LoginScreen` have been updated for testing purposes.

dinhthongchau committed:
Refactor: Implement multi-user transaction management with persistence

This commit refactors the `TransactionRepository` to support multiple users and persist transaction data using `SharedPreferences`.

Key changes:
- **Multi-user support:**
    - Transactions are now stored in a map (`_userTransactions`) keyed by user email.
    - The repository listens to `FirebaseAuth.authStateChanges` to load/save data for the current user.
    - `_getCurrentUserIdentifier()` is used to get the current user's email.
- **Data persistence:**
    - `_loadSavedData()` loads all users' transaction data from `SharedPreferences` on startup.
    - `_loadDataForUser(email)` loads transactions for a specific user.
    - `_saveDataForUser(email)` saves transactions for a specific user.
    - Transactions are serialized to JSON for storage. When loading, categories are matched to reconstruct `TransactionModel` instances.
- **Mock data generation:**
    - `generateMockData(userEmail)` now generates mock data specific to a user, incorporating the user's email into transaction titles and amounts for easier identification.
    - Mock data is only generated for a specific test user (`nkhiet@gmail.com`) if no saved data exists for them. Other users will have an empty transaction list if no data is found.
- **Cache management:**
    - `clearUserCache(email)` clears in-memory data for a specific user.
    - `clearAllData()` clears all in-memory and persisted transaction data.
- **Repository lifecycle:**
    - The repository is now a singleton (`TransactionRepository._instance`).
    - It initializes by loading saved data and listening to auth state changes.
- **Logging:**
    - Added `_log()` helper for debug messages, prefixed with "📊 TransactionRepo:".
- **API methods (`getTransactionsAPI`, `updateTransaction`, `addTransaction`, `deleteTransaction`)**:
    - Now operate on the current user's transaction list.
    - Changes are persisted to `SharedPreferences` after modifications.
- **Public `transactionData` getter:**
    - Returns the transaction list for the current user.

dinhthongchau committed:
Refactor: Store transactions in Firestore instead of SharedPreferences

This commit refactors the `TransactionRepository` to use Firestore for storing and retrieving transaction data, replacing the previous SharedPreferences-based implementation.

Key changes:
- **Firestore Integration:**
    - Transactions are now stored under `/transactions/users/{userEmail}/user_transactions/items/{document_id}`.
    - `_loadDataForUser` fetches transactions from Firestore.
    - `_saveTransactionToFirestore` is a new private method to save individual transactions to Firestore.
    - `addTransaction`, `updateTransaction`, and `deleteTransaction` now interact with Firestore.
- **Data Loading and Initialization:**
    - The logic for loading data on user change (`FirebaseAuth.instance.authStateChanges()`) has been updated to use Firestore.
    - The concept of "initialized users" (`_initializedUsers`) is maintained to avoid redundant Firestore reads.
- **Mock Data Generation:**
    - `generateMockData` now generates transactions from January 1, 2023, to the current date, creating 2-3 random transactions per day.
    - Generated mock data is also saved to Firestore if the user is `nkhiet3@gmail.com`.
- **Removed SharedPreferences Logic:**
    - All code related to `SharedPreferences` for storing transactions has been removed (`_loadSavedData`, `_saveDataForUser`, etc.).
    - The `clearAllData` method now clears data from Firestore for the current user.
- **Profile Screen:**
    - Removed the explicit call to `TransactionRepository().saveDataForUser(email)` on logout, as data is now persisted to Firestore more dynamically.
- **Home Screen UI:**
    - Ensured `Text` widgets for title and amount in `_buildCardInfo` use `Expanded` to prevent overflow issues with long text.
- **README Update:**
    - Added a note about Firestore rules for the `transactions` collection and included an image `img.png` illustrating these rules.

dinhthongchau committed:
Refactor: Enhance transaction list performance and update mock data

This commit introduces several changes to improve performance and data generation:

**Performance Optimization:**
- Refactored `HomeScreenMixin` and `TransactionScreenMixin` to use `ListView.builder` for rendering transaction lists. This optimizes performance by only building visible items, improving scroll smoothness and reducing memory usage, especially for large datasets.
- Simplified transaction grouping logic in `TransactionScreenMixin` by using `putIfAbsent`.

**Mock Data Generation:**
- Updated `TransactionRepository.generateMockData` to create more realistic and extensive transaction data:
    - Generates ~35 transactions for 2023 (monthly salary, ~23 expenses, ~11 savings).
    - Generates ~45 transactions for 2024 (monthly salary, ~33 expenses, ~12 savings).
    - Generates 1-3 daily transactions (expense/save) and monthly salary for 2025 up to the current date.
- This provides a richer dataset for testing and development.
- Removed the initial emission of `TransactionInitial` state in `TransactionBloc`'s `_onLoadTransactions` to prevent potential redundant UI updates.

**Configuration:**
- Updated the default email in `LoginScreen` to `test@gmail.com`.
- Updated `google-services.json` with new Firebase project configuration.

ThinhSuy1 committed:
Refactor code, using assets generator

dinhthongchau committed:
Refactor: Enhance transaction loading and caching logic

This commit refactors the transaction loading and caching mechanisms:

**TransactionBloc:**
- Implements debouncing for `UserChangedEvent` to prevent multiple rapid reloads when the user authentication state changes frequently.
- Ensures that transactions are only reloaded if the user ID has actually changed, avoiding unnecessary reloads for the same user.

**TransactionRepository:**
- Introduces a 5-minute cache (`cacheDuration`) for loaded transactions.
- Transactions will only be fetched from Firestore if the cache is invalid (expired or user has changed) or if the local cache is empty for the current user.
- Logs cache status (valid/invalid) for better debugging.
- Removed the `generateMockData` and `_getCategory` methods as they are no longer needed for current functionality and were primarily for development/testing.

**AddTransactionScreen:**
- Navigates back (`context.pop()`) after successfully adding a transaction.

**BottomNavigationBarScaffold:**
- Refines transaction loading logic: transactions are now explicitly loaded only when navigating to the transaction tab (index 2) and it wasn't the previously active tab. This prevents redundant loading when switching between other tabs.
- Ensures `LoadingUtils.showLoading` is always called within `addPostFrameCallback` for loading and error states.

**LoginScreen & SignUpScreen:**
- Updated default/test input values for email, password, name, and DOB for easier testing.

dinhthongchau committed:
Merge branch 'features/authentication-profile' of https://github.com/Jaycii2808/finance-management into features/authen+trans+2905

# Conflicts:
#	lib/data/model/user_model.dart
#	lib/main.dart
#	lib/presentation/bloc/bloc_observe.dart
#	lib/presentation/bloc/notification/notification_bloc.dart
#	lib/presentation/bloc/notification/notification_event.dart
#	lib/presentation/bloc/notification/notification_state.dart
#	lib/presentation/routes.dart
#	lib/presentation/screens/home/home_screen.dart
#	lib/presentation/screens/login/login_screen.dart
#	lib/presentation/screens/notification/notification_screen.dart
#	pubspec.yaml

dinhthongchau committed:
Refactor: Remove unused imports and Hive dependency

This commit removes several unused imports across various files, including:
- `app_colors.dart` from authentication screens and `new_password_screen.dart`.
- `user_model.dart` and `hive.dart` from `sign_up_screen.dart`.
- `user_model.dart` from `add_transaction_screen_mixin.dart`.
- Several BLoC related imports from `profile_screen.dart`.

The dependency on `hive` and `shared_preferences` has been removed from `pubspec.yaml` as they are no longer used.

Additionally, the test code related to Hive in `SignUpScreen` has been removed.

Minor UI adjustments:
- Replaced `Colors.black.withOpacity(0.3)` with `Colors.black.withValues(alpha: 0.3)` in `login_screen.dart`.
- Removed `Key? key` and `super(key: key)` from constructors in `profile_online_support_ai_screen.dart`, `chat_input_bar.dart`, and `chat_bubble.dart`, using `super.key` instead where appropriate.
- Removed unused `background` and `onBackground` from `theme_dark.dart`.

dinhthongchau committed:
feat: Add shared_preferences dependency

This commit introduces the `shared_preferences` package (version ^2.5.3) to the project. This package will be used for persistent storage of simple data.

dinhthongchau committed:
Refactor: Restructure project folders and update imports

This commit introduces a significant restructuring of the project's folder organization, primarily by:

- **Deleting shared export files:**
    - `lib/presentation/widgets/shared_folders/shared_blocs.dart`
    - `lib/presentation/widgets/shared_folders/shared_model.dart`
    - `lib/presentation/widgets/shared_folders/shared_repositories.dart`
    - `lib/presentation/widgets/shared_folders/shared_screens.dart`
    - `lib/presentation/widgets/shared_folders/shared_utils.dart`
    - `lib/presentation/widgets/shared_folders/shared_widgets.dart`
    - `lib/presentation/widgets/shared_folders/shared_data.dart`
- **Moving models to more specific directories:**
    - `lib/data/model/user_model.dart` was deleted (appears to be unused or its functionality moved elsewhere).
    - `lib/data/model/transaction_model.dart` -> `lib/data/model/transaction/transaction_model.dart`
    - `lib/data/model/calendar_chart_model.dart` -> `lib/data/model/analysis/calendar_chart_model.dart`
    - `lib/data/model/category_model.dart` -> `lib/data/model/transaction/category_model.dart`
    - `lib/data/model/notification_model.dart` -> `lib/data/model/notification/notification_model.dart`
    - `lib/data/model/analysis_model.dart` -> `lib/data/model/analysis/analysis_model.dart`
- **Moving repositories to a unified `repositories` directory:**
    - `lib/data/repository/user/user_repository.dart` -> `lib/data/repositories/user_repository.dart`
    - `lib/data/repository/help_faqs/help_faqs_repository.dart` -> `lib/data/repositories/help_faqs_repository.dart`
- **Moving generated assets:**
    - `lib/gen/assets.gen.dart` -> `lib/data/gen/assets.gen.dart`
- **Updating import paths:** All affected files have had their import statements updated to reflect the new file locations.
- **Consolidating common exports:** The `lib/presentation/shared_data.dart` file now serves as a central point for exporting commonly used models, screens, BLoCs, utils, and widgets.
- **Standardizing route names:** Route names in `lib/presentation/routes.dart` are now defined as static constants within their respective screen files for better organization and type safety (e.g., `NewPasswordScreen.routeName`).
- **Minor code cleanup:** Removed commented-out code and unused comments in various files (e.g., `ProfileEditScreen`, `ProfileOnlineSupportAiScreen`).

These changes aim to improve the project's maintainability, readability, and overall structure by promoting a more feature-oriented and organized file system.

dinhthongchau committed:
Refactor: Update Android app label

The application label in `AndroidManifest.xml` has been changed from "finance_management" to "Finance Management V1". This provides a more user-friendly and descriptive name for the application.

dinhthongchau committed:
feat: Integrate Firebase for Web and update configurations

This commit introduces Firebase support for the web platform and includes several configuration updates and minor fixes.

**Firebase for Web:**
- Added `firebase.json` to define Firebase configurations for Android and Web.
- Created `lib/data/services/firebase_options.dart` containing platform-specific Firebase options.
- Updated `web/index.html` to include Firebase JavaScript SDKs and initialization script for web.
- Modified `main.dart` to initialize Firebase using `DefaultFirebaseOptions.currentPlatform`.

**Configuration and Permission Handling:**
- Updated `android/settings.gradle.kts` to include the `google-services` plugin.
- Modified `lib/data/repositories/user_repository.dart`:
    - `requestStoragePermission()` now returns `true` for web as permissions are not required.
    - `editUser()` now skips avatar upload for web platform.

**UI and Logic Updates:**
- `AnalysisScreen`: Added loading indicator handling.
- `BottomNavigationBarScaffold`:
    - Added logic to load analysis data when the analysis tab (index 1) is selected.
- `LoginScreen`:
    - Updated default login credentials.
    - Improved error handling and logging during login.
- `customPrint` function in `lib/core/utils/common_functions.dart` now prefixes web logs with "WEB: ".

**File Renaming:**
- Renamed `lib/utils/notification_helper.dart` to `lib/core/utils/notification_helper.dart` and updated import paths.

dinhthongchau committed:
Refactor: Introduce SharedLayout for responsive padding and update navigation

This commit introduces `SharedLayout.dart` to provide responsive screen padding based on screen width (web vs. mobile). This utility is now used across various screens to ensure consistent layout.

**Key changes:**

- **New `SharedLayout.dart`:**
    - Added `getScreenPadding(BuildContext context)` static method that returns `EdgeInsets.symmetric(horizontal: 150)` for web (width > 600) and `EdgeInsets.zero` for mobile.
- **Screen padding updates:**
    - Applied `SharedLayout.getScreenPadding` to the body of the following screens:
        - `AnalysisScreen`
        - `CalendarScreen`
        - `AddTransactionScreen`
        - `CategoryDetailSaveScreen`
        - `CategoryDetailScreen`
        - `CategoriesScreen` (Categories Home)
        - `CategoryListScreen`
        - `HomeScreen`
        - `LoginScreen`
        - `NotificationScreen`
        - `ProfileScreen`
        - `SearchScreen`
        - `SignUpScreen`
        - `TransactionScreen`
- **Navigation updates:**
    - Changed `context.go` to `context.push` for navigating to `ForgetPasswordScreen` from `SplashScreen` and `LoginScreen`.
    - Set `LoginScreen` as the initial route (`/`) in `routes.dart`.
- **UI adjustments:**
    - **`BottomNavigationBarScaffold`:**
        - Implemented a `LayoutBuilder` to switch between a bottom navigation bar (mobile) and a left-side navigation bar (web, maxWidth >= 800).
        - The left navigation bar is displayed in web layout with a width of 100 and `AppColors.lightGreen` background.
    - **`ProfileScreen`:**
        - Replaced `ProfileAppBar` with the shared `buildHeader` widget.
        - Encapsulated the main body content into a `buildBodyProfile` method.
        - Removed `SafeArea` wrapper as padding is handled by `SharedLayout`.
    - **`AnalysisScreen`:** Added `SizedBox(height: 10)` at the beginning of the `Column` in `buildBody` in `AnalysisScreenMixin`.

These changes aim to improve the responsiveness and consistency of the application's UI across different screen sizes.

dinhthongchau committed:
Refactor: Adjust grid layout and navigation logic

This commit introduces the following changes:

- **Login Screen:**
    - Modified navigation logic after successful login. Users identified as `helper` are now redirected to `ProfileOnlineSupportHelperScreen`, while other users are directed to `HomeScreen`.

- **Category List Screen & Analysis Screen:**
    - Updated the `crossAxisCount` for `SliverGridDelegateWithFixedCrossAxisCount` to be responsive to screen width.
        - On screens wider than 600 pixels, the grid will display more columns (6 for categories, 5 for analysis items).
        - On smaller screens, the default column count (3 for categories, 2 for analysis items) remains.

dinhthongchau committed:
feat: Add Firebase web, DevTools, and launcher icon configurations

This commit introduces several configuration files and updates related to Firebase, Flutter DevTools, and application launcher icons.

**Firebase:**
- Added `.firebaserc` to set the default Firebase project.
- Added `firebase.json` with hosting configuration for the "biwofinance" site, deploying the `build/web` directory.
- Added `public/index.html` as a placeholder for Firebase Hosting.
- Added `.firebase/hosting.YnVpbGRcd2Vi.cache` and `.firebase/hosting.cHVibGlj.cache` which appear to be Firebase hosting cache files.
- Updated `android/settings.gradle.kts` to use `com.google.gms.google-services` version `4.4.2`.

**Flutter DevTools:**
- Added `devtools_options.yaml` to store settings for Dart & Flutter DevTools.

**Launcher Icons:**
- Added `flutter_launcher_icons.yaml` to configure launcher icons for Android and Web.
- Added `assets/icon/icon.png` as the source image for launcher icons.
- Added the `flutter_launcher_icons: ^0.14.3` dependency to `pubspec.yaml`.
- Generated Android launcher icons in `android/app/src/main/res/mipmap-*` directories.
- Updated `android/app/src/main/AndroidManifest.xml` to use the new `launcher_icon` and changed the app label to "BiwoFinance V1".
- Updated `web/favicon.png` and web manifest icons.
- Updated `web/manifest.json` with new `background_color` and `theme_color`.

**Other Changes:**
- Updated `web/index.html` title to "BiwoFinance V1".
- Updated `android/app/build.gradle.kts` to enable `-Xlint:deprecation` for Kotlin compiler warnings.

dinhthongchau committed:
Refactor: Standardize date format to dd/MM/yyyy

This commit updates the date formatting across various screens to use the `dd/MM/yyyy` pattern.

**Affected screens and components:**

- `CalendarScreenMixin`: Updated transaction date display.
- `AddTransactionScreenMixin`: Changed date format for notification time and date input field.
- `TransactionReusedWidgets`:
    - Standardized date format for notification time.
    - Replaced `TextFormField` for date input with a `GestureDetector` that opens a `showDatePicker` dialog for a more user-friendly date selection experience.
- `HomeScreenMixin`: Updated date format for displayed transaction dates.
- `SearchScreenMixin`: Modified date format for selected date display and transaction item time.

Jaycii2808 commented on A.java:
Delete this

Jaycii2808 commented on README.md:
Refactor readme

Jaycii2808 commented on lib/core/utils/notification_helper.dart:
```diff
@@ -0,0 +1,43 @@
+          final chatRoomId = response.payload;
+          // Navigate to chat screen with chatRoomId
+          router.go(
+            '/profile-online-support-helper-center?chatRoomId=$chatRoomId',
```
Fix hardcore route

Jaycii2808 commented on lib/data/services/firebase_chat_service.dart:
remove cmt vi

Jaycii2808 commented on lib/presentation/screens/analysis/analysis/analysis_screen.dart:
```diff
@@ -0,0 +1,105 @@
+            appBar: buildHeader(
+              context,
+              "Analysis",
+              "/home-screen/notifications-screen",
```
???

Jaycii2808 commented on lib/presentation/screens/analysis/analysis/analysis_screen.dart:
```diff
@@ -0,0 +1,105 @@
+          appBar: buildHeader(
+            context,
+            "Analysis",
+            "/home-screen/notifications-screen",
```
???

Jaycii2808 commented on lib/presentation/screens/authentication/forget_password/new_password_screen.dart:
```diff
@@ -217,7 +217,9 @@ class _NewPasswordScreenState extends State<NewPasswordScreen> {
                                                 .isNotEmpty
                                         ? () {
-                                          context.go('/password-changed-splash');
+                                          context.go(
```
Refactor -> Route name

ThinhSuy1 committed:
Fix and refactor code following Jaycii's comments

ThinhSuy1 committed:
fix bug resume app, manage life cycle

ThinhSuy1 committed:
Fix the bug, disable android back button in initial screen

dinhthongchau committed:
Refactor: Enhance AnalysisBloc with date range functionality and UI improvements

This commit introduces several enhancements to the Analysis feature:

**AnalysisBloc & State:**
- Added `startDate` and `endDate` to `AnalysisState` to manage custom date ranges.
- Introduced `ChangeDateRangeEvent` to allow users to select custom date ranges for analysis.
- Modified `_onLoadAnalysisData` and `_onChangeTimeFilter` to calculate and use default date ranges based on the selected `TimeFilterAnalysis`:
    - **Daily:** Shows the current week (Monday to current day).
    - **Weekly:** Shows the current month, broken down into 4 weeks.
- Updated `_generateDailyData` and `_generateWeeklyData` to accept and use `startDate` and `endDate` for filtering transactions.
    - `_generateDailyData` now always shows a full week (Monday to Sunday) based on the `startDate`.
    - `_generateWeeklyData` now divides the month (defined by `startDate`) into four distinct weeks (1-7, 8-14, 15-21, 22-end of month).

**AnalysisScreen & Mixin:**
- Added `_buildDateRangeSelector` to the `AnalysisScreenMixin` to display and allow navigation/selection of date ranges.
    - For "Daily" and "Weekly" filters, users can navigate by week/month and open a date/month picker.
    - Navigation is restricted to prevent going beyond the current day/month.
- Implemented `_navigateDateRange` to handle date range changes when arrow buttons are pressed.
- Implemented `_showDateRangePicker` to display a `showDateRangePicker` (for Daily) or a custom `showMonthPicker` (for Weekly).
- Created a custom `showMonthPicker` dialog for selecting a specific month and year.
- Ensured `AnalysisBloc` is only called if `context.mounted` in `BottomNavigationBarScaffold`.

**UI & Other Changes:**
- `HomeScreen`: Removed the unused `label` parameter.
- `AddTransactionScreenMixin`: Implemented debouncing for text input fields to improve performance and user experience.
- `TransactionReusedWidgets`:
    - Adjusted `dismissThresholds` for `Dismissible` transaction items.
    - Added rounded corners to the background of `Dismissible` items.
    - Wrapped `Dismissible` child and `AlertDialog` content with `AnimatedOpacity` for smoother transitions (though opacity is currently static at 1.0).
- `SharedLayout`: Adjusted responsive padding breakpoints.
- `AppColors`: Added `lightGrey`.
- Updated route paths in `ProfileScreen`, `AnalysisScreen`, `TransactionScreen`, and `routes.dart` to use static route names for consistency.
- Deleted placeholder file `A.java`.
- Deleted default Firebase Hosting `index.html`.

dinhthongchau committed:
Refactor: Enhance Bloc state management and integrate Gemini API key via .env

This commit introduces several improvements across various BLoCs and integrates the Gemini API key using `flutter_dotenv`.

**Bloc State Enhancements:**
- Implemented `props` getter in the following BLoC states for improved equatability and testing:
    - `NotificationState` (Initial, Loading, Success, Error)
    - `SearchState` (Initial, Loading, Success, Error)
    - `HomeState` (Initial, Loading, Success, Error)
    - `AnalysisState` (Initial, Loading, Success, Error)
    - `CategoryState` (Initial, Loading, Success, Error)
    - `CalendarState` (Initial, Loading, Success, Error)
    - `TransactionState` (Initial, Loading, Success, Error)
    - `AddTransactionState` (Initial, Loading, Success, Error)
    - `UserState` (Initial, Loading, Deleted)

**Gemini API Integration:**
- Modified `ProfileOnlineSupportAiScreen` to fetch the Gemini API key from environment variables using `flutter_dotenv`.
- Added `assets/dotenv.example` as a template for the `.env` file.
- Included `assets/dotenv` in `pubspec.yaml` and loaded it in `main.dart`.
- Updated `.gitignore` to exclude `.env` and other generated/sensitive files.

**Localization & UI Text:**
- Changed some error messages and UI text from Vietnamese to English in:
    - `ProfileOnlineSupportAiScreen`
    - `CategoryListScreenMixin`
    - `BottomNavigationBarScaffold`

**Minor Refactoring:**
- Removed unused `isLoading` parameter from `LoginButton` in `LoginScreen`.
- Deleted an apparently unused image file `img.png`.

ThinhSuy1 committed:
add back from support ai screen to lobby screen, remove loading indicator when login, write README.md

ThinhSuy1 committed:
Merge branch 'features/firebase-finance-with-refactor-authen' of https://github.com/Jaycii2808/finance-management into features/firebase-finance-with-refactor-authen

ThinhSuy1 committed:
new README.md

dinhthongchau committed:
Docs: Update README.md and add project screenshot

This commit updates the `README.md` file to:
- Remove "Google, Facebook supported" from login features.
- Remove "Modern UI with light/dark theme support" from features.
- Remove the "Project Structure" section.
- Remove the "Contribution & Development" section.
- Simplify the "Notes" section by removing the point about Bloc for global state management.

Additionally, a new image file `img.png` has been added, presumably a screenshot of the project.

dinhthongchau commented:
@Jaycii2808 hi

dinhthongchau commented:
@Jaycii2808
